server/services/lists/comCaseloadService.ts

```diff
@@ -348,4 +376,102 @@ export default class ComCaseloadService {
       'licence status is not one of ACTIVE, VARIATION_IN_PROGRESS, VARIATION_SUBMITTED, VARIATION_APPROVED, VARIATION_REJECTED, REVIEW_NEEDED'
     )
   }
+
+  createCaseloadViewModel = (caseload: ManagedCase[]): ComCase[] => {
+    return caseload
+      .map(c => {
+        const licence = this.findLicenceToDisplay(c)
+        const releaseDate = c.nomisRecord.releaseDate || c.nomisRecord.conditionalReleaseDate
+        const sortDate = releaseDate && parseIsoDate(releaseDate)
+        const { hardStopDate, hardStopWarningDate } = licence
+        return {
+          name: convertToTitleCase(`${c.nomisRecord.firstName} ${c.nomisRecord.lastName}`.trim()),
+          crnNumber: c.deliusRecord.offenderCrn,
+          prisonerNumber: c.nomisRecord.prisonerNumber,
+          releaseDate: releaseDate && format(releaseDate, 'dd/MM/yyyy'),
+          sortDate,
+          licenceId: licence.id,
+          licenceStatus: licence.status,
+          licenceType: licence.type,
+          probationPractitioner: c.probationPractitioner,
+          hardStopDate: hardStopDate && format(hardStopDate, 'dd/MM/yyyy'),
+          hardStopWarningDate: hardStopWarningDate && format(hardStopWarningDate, 'dd/MM/yyyy'),
+          kind: licence.kind,
+          isDueForEarlyRelease: c.cvlFields?.isDueForEarlyRelease,
+          licenceCreationType: licence.licenceCreationType,
+        }
+      })
+      .sort((a, b) => {
+        return (a.sortDate?.getTime() || 0) - (b.sortDate?.getTime() || 0)
+      })
+  }
+
+  createVaryCaseloadViewModel = (caseload: ManagedCase[]): ComCase[] => {
+    return caseload.map(managedCase => {
+      const licences = managedCase.licences.filter(l => l.status !== LicenceStatus.TIMED_OUT)
+      const licence =
+        licences.length > 1
+          ? licences.find(l => l.status !== LicenceStatus.ACTIVE && l.status !== LicenceStatus.REVIEW_NEEDED)
+          : _.head(licences)
+
+      const releaseDate = managedCase.nomisRecord.releaseDate || managedCase.nomisRecord.conditionalReleaseDate
```

Comment:
**suggestion:** As above, I think this should come directly from the licence and probably *shouldn't* fall back to NOMIS (though happy to discuss this)
This could mean that the releaseDate is null, but would also highlight that dates are missing in CVL, which we would need to fix!
Thinking on it further, IS91 and remand cases (though remand are currently OOS) would have no dates in NOMIS but would (hopefully) have dates in CVL, so we'd definitely want to be at least prioritising the licence, if not only reading from it.